crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -339,6 +334,85 @@ impl SequencesState {
     }
 }
 
+/// Utility to query the system tables and return their concrete table row
+pub struct SystemTableQuery<'a> {
+    db: &'a Inner,
+    ctx: &'a ExecutionContext<'a>,
+}
+
+impl<'a> SystemTableQuery<'a> {
```

Comment:
I think these should come with a comment about their performance. Each of these functions requires materializing the full table into a vec.

Reply:
So far this is only for simplify the testing. Probably add a `todo` to change them when we have a better mem-arch?